File: platforms/ios/Dezel/Dezel/View/Type/Overscroll.swift
/**
 * @enum Overscroll
 * @since 0.2.0
 */
public enum Overscroll {

	case auto
	case never
	case always
	case alwaysX
	case alwaysY

	/**
	 * @method get
	 * @since 0.2.0
	 */
	public static func get(_ value: JavaScriptProperty) -> Overscroll {

		switch (value.type) {

			case .number:
				return self.get(value: value.number)
			case .string:
				return self.get(value: value.string)
			case .boolean:
				return self.get(value: value.boolean)

			default:
				return .auto
		}
	}

	/**
	 * @method get
	 * @since 0.2.0
	 */
	public static func get(value: String) -> Overscroll {

		switch (value) {

			case "auto":
				return .auto
			case "never":
				return .never
			case "always":
				return .always
			case "always-x":
				return .alwaysX
			case "always-y":
				return .alwaysY

			default:
				return .auto
		}
	}

	/**
	 * @method get
	 * @since 0.2.0
	 */
	public static func get(value: Double) -> Overscroll {
		return value == 0.0 ? .never : .auto
	}

	/**
	 * @method get
	 * @since 0.2.0
	 */
	public static func get(value: Bool) -> Overscroll {
		return value == true ? .auto : .never
	}
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/view/type/Overscroll.kt
package ca.logaritm.dezel.view.type

import ca.logaritm.dezel.core.JavaScriptProperty
import ca.logaritm.dezel.core.JavaScriptPropertyType

/**
 * @enum Overscroll
 * @since 0.2.0
 */
public enum class Overscroll {

	AUTO,
	NEVER,
	ALWAYS,
	ALWAYS_X,
	ALWAYS_Y;

	companion object {

		/**
		 * @method get
		 * @since 0.2.0
		 */
		public fun get(value: JavaScriptProperty): Overscroll {
			return when (value.type) {
				JavaScriptPropertyType.BOOLEAN -> this.get(value.boolean)
				JavaScriptPropertyType.NUMBER  -> this.get(value.number)
				JavaScriptPropertyType.STRING  -> this.get(value.string)
				else                           -> Overscroll.AUTO
			}
		}

		/**
		 * @method get
		 * @since 0.2.0
		 */
		public fun get(value: String): Overscroll {
			return when (value) {
				"auto"     -> Overscroll.AUTO
				"never"    -> Overscroll.NEVER
				"always"   -> Overscroll.ALWAYS
				"always-x" -> Overscroll.ALWAYS_X
				"always-y" -> Overscroll.ALWAYS_Y
				else       -> Overscroll.AUTO
			}
		}

		/**
		 * @method get
		 * @since 0.2.0
		 */
		public fun get(value: Double): Overscroll {
			return if (value == 0.0) Overscroll.NEVER else Overscroll.AUTO
		}

		/**
		 * @method get
		 * @since 0.2.0
		 */
		public fun get(value: Boolean): Overscroll {
			return if (value == true)  Overscroll.AUTO else Overscroll.NEVER
		}
	}
}